src/net/server/stream.rs

```diff
@@ -0,0 +1,667 @@
+//! Support for stream based server transports.
+//!
+//! Wikipedia defines [stream] as:
+//!
+//! > _A reliable byte stream is a common service paradigm in computer
+//! > networking; it refers to a byte stream in which the bytes which emerge
+//! > from the communication channel at the recipient are exactly the same,
+//! > and in exactly the same order, as they were when the sender inserted
+//! > them into the channel._
+//! >
+//! > _The classic example of a reliable byte stream communication protocol is
+//! > the **Transmission Control Protocol**, one of the major building blocks of
+//! > the Internet._
+//!
+//! [stream]: https://en.wikipedia.org/wiki/Reliable_byte_streamuse
+use arc_swap::ArcSwap;
+use core::future::poll_fn;
+use core::ops::Deref;
+use core::sync::atomic::{AtomicUsize, Ordering};
+use core::time::Duration;
+use octseq::Octets;
+use std::fmt::Debug;
+use std::io;
+use std::net::SocketAddr;
+use std::string::{String, ToString};
+use std::sync::{Arc, Mutex};
+use tokio::net::TcpListener;
+use tokio::sync::watch;
+use tokio::time::{interval, timeout, MissedTickBehavior};
+use tracing::{error, trace, trace_span};
+
+use crate::net::server::buf::BufSource;
+use crate::net::server::error::Error;
+use crate::net::server::metrics::ServerMetrics;
+use crate::net::server::middleware::chain::MiddlewareChain;
+use crate::net::server::service::Service;
+use crate::net::server::sock::AsyncAccept;
+use crate::utils::config::DefMinMax;
+
+use super::buf::VecBufSource;
+use super::connection::{self, Connection};
+use super::service::ServerCommand;
+use crate::base::wire::Composer;
+
+// TODO: Should this crate also provide a TLS listener implementation?
```

Comment:
No.

Reply:
So, to expand on this a little now that I have more time: `examples/server-transport.rs` shows how to do TLS with the servers provided by this PR, but this PR does not add such TLS support directly in the library itself. The reason for this is that TLS could be via RustLS or OpenSSL or something else and I wanted to avoid adding that complexity to the library, but am still unsure if this is the right approach hence the comment.

Since writing that TODO comment I have also noticed that the client-transport code handles TLS a little differently and I haven't yet had time to review if that is due to the different nature of client vs server, or if it's a pattern that may make sense to use for TLS for the server as well.